components/suggest/src/db.rs

```diff
@@ -306,6 +306,13 @@ impl<'a> SuggestDao<'a> {
                         let cooked_url = cook_raw_suggestion_url(&raw_url);
                         let raw_click_url = row.get::<_, String>("click_url")?;
                         let cooked_click_url = cook_raw_suggestion_url(&raw_click_url);
+                        let icon = match row.get("icon_content")? {
+                            Some(content) => Some(SuggestionIcon {
+                                content,
+                                mime_type: row.get("icon_mimetype")?,
+                            }),
+                            _ => None,
+                        };
```

Comment:
Nit: I would write this using `Option::map()`, but I'm not sure it's better and Lina might disagree:

```rust
                        let icon_content = row.get::<_, Option<_>>("icon_content")?;
                        let icon_mimetype = row.get::<_, Option<_>>("icon_mimetype")?;
                        let icon = icon_content.map(|content| SuggestionIcon {
                            content,
                            mime_type: icon_mimetype.unwrap_or_else(|| "".into()),
                        });
```